src/_guides/openapi/advanced-ref-usage.md

```diff
@@ -0,0 +1,335 @@
+
+Others hide their OpenAPI and AsyncAPI by choice for security reasons, but that's never made much sense because Stripe, PayPal, Box, GitHub, and plenty of other massive API companies have their API descriptions out in public and nobody has hacked them. APIs should be protected with firewalls and API keys, but OpenAPI and AsyncAPI information can be plastered all over the place. Another vote for the public static site.
+
+There is an argument for making public APIs public and keeping internal API's private, and some hosted API documentation tools can help with that, or you can host internal API docs on a different static site that's only available on the company network. Either way you'll need to keep your public APIs public, and keep your shared components public, then hide the internal APIs that reference those. That gives you the best of both worlds.
+
+## Propagating Changes
+
+Using tools like Bump.sh you get all the benefits of a tool that understands $ref, but without any of the hassle of needing to bundle documents up. 
+
+Like any tool which uses a build step, this has the pro and the con of meaning that documentation is built at a certain point in time. Changes that happen to the $ref'ed resources - whether they're in another repository, or being pulled in via URL - will take some time to appear in your API.
+
+For example, if the Widget API is using a shared Company schema via `$ref: https://widgets.com/schema/company.json`, and company decides to add VAT number as a property, your Widget API documentation is not going to mention that property until your next build.
+
+Is that a good thing or a bad thing? It can be both depending on the scenario, but having changes appear in your API without your knowledge is probably not ideal.
+
+## Tools for Bundling & Splitting 
+
+Bundling is usually only needed if you are working with older or strange tools which do not support $ref properly (or at all). If you are working with Bump.sh CLI you won’t need to bundle, but if a tool wants you to import a single `openapi.yaml` document you might need to bundle.
+
+```
+$ redocly bundle openapi.yaml -o openapi-bundled.yaml
+bundling openapi.yaml...
+📦 Created a bundle for openapi.yaml at openapi-bundled.yaml 105ms.
+```
+
+This will grab all of the `$ref`'s that use "external files" or URLs and move the contents into the relevant subsection of `components` in the `openapi-bundled.yaml` document.
+
+Splitting does the opposite. If you have a massive painful document (maybe generated from HTTP or converted from Postman) you can split it down into multiple documents with a sensible folder structure, ditch the original, commit all that to Git, and push it up to Bump.sh with all the `$ref`'s intact.
+```
+redocly split generated-openapi.json --outDir api/
+
+bump deploy api/
```

Comment:
This won't work unfortunately (passing a directory to deploy a SINGLE documentation), you'll need to point the deploy command to the “root” OpenAPI file, what is the default name of the openapi document after the `redocly split` command? (If I understand [their docs](https://redocly.com/docs/cli/commands/split/#example) it seems to be `openapi.[format]`)

```suggestion
bump deploy api/openapi.json
```